In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
%matplotlib inline

In [3]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv' )
gdp_df.head(6)

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2020  1970.560169              NaN
1     Afghanistan  2019  2065.036235              NaN
2     Afghanistan  2018  2033.804389              NaN
3     Afghanistan  2017  2058.400221              NaN
4     Afghanistan  2016  2057.067978              NaN
5     Afghanistan  2015  2068.265904              NaN

In [12]:
internet_df = pd.read_csv('../data/internet_use.csv')
internet_df.head(6)

Country or Area  Year     Value  Value Footnotes
0     Afghanistan  2014  6.390000              NaN
1     Afghanistan  2013  5.900000              NaN
2     Afghanistan  2012  5.454545              NaN
3     Afghanistan  2011  5.000000              NaN
4     Afghanistan  2010  4.000000              NaN
5     Afghanistan  2009  3.550000              NaN

Number of rows and columns

In [5]:
gdp_df.shape

(7176, 4)

There are 7176 rows and 4 columns in gdp dataframe

In [6]:
internet_df.shape

(4674, 4)

There are 4675 rows and 4 columns in internet_use dataframe

Show datatypes of colums

In [7]:
print(type(gdp_df['Country or Area']))
print(type(gdp_df.Year))
print(type(gdp_df.Value))
print(type(gdp_df['Value Footnotes']))

<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>


Show data types of columns:

In [8]:
gdp_df.columns

Index(['Country or Area', 'Year', 'Value', 'Value Footnotes'], dtype='object')

In [9]:
internet_df.columns


Index(['Country or Area', 'Year', 'Value', 'Value Footnotes'], dtype='object')

In [62]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7176 entries, 0 to 7175
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  7176 non-null   object 
 1   Year             7176 non-null   int64  
 2   Value            7176 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 224.4+ KB


In [63]:
internet_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4676 entries, 0 to 4675
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  4676 non-null   object 
 1   Year             4676 non-null   object 
 2   Value            4495 non-null   float64
 3   Value Footnotes  948 non-null    float64
dtypes: float64(2), object(2)
memory usage: 146.2+ KB


Different datatypes between the two DFs. gdp_df: obj, int64, float64, float64. internet_df: object, object, float64, float64.

Drop Column 'Value Footnote'

In [21]:
gdp_three = gdp_df.drop(columns = ['Value Footnotes'])
internet_three = internet_df.drop(columns = ['Value Footnotes'])
gdp_three.head()
internet_three.head()

Country or Area  Year     Value
0     Afghanistan  2014  6.390000
1     Afghanistan  2013  5.900000
2     Afghanistan  2012  5.454545
3     Afghanistan  2011  5.000000
4     Afghanistan  2010  4.000000

Change the columns for the GDP Per Capita DataFrame to ‘Country’, ‘Year’, and ‘GDP_Per_Capita’.

In [32]:
gdp_three.columns = ['Country','Year', 'GDP_Per_Capita']
internet_three.columns = ['Country','Year', 'Internet_Users_Pct']
gdp_three.head()
internet_three.head()

Country  Year  Internet_Users_Pct
0  Afghanistan  2014            6.390000
1  Afghanistan  2013            5.900000
2  Afghanistan  2012            5.454545
3  Afghanistan  2011            5.000000
4  Afghanistan  2010            4.000000

Merge two dataframes as gdp_and_internet_use

In [53]:
gdp_and_internet_use = pd.merge(gdp_three, internet_three, on=['Country'])
gdp_and_internet_use

Country  Year_x  GDP_Per_Capita Year_y  Internet_Users_Pct
0       Afghanistan    2020     1970.560169   2014            6.390000
1       Afghanistan    2020     1970.560169   2013            5.900000
2       Afghanistan    2020     1970.560169   2012            5.454545
3       Afghanistan    2020     1970.560169   2011            5.000000
4       Afghanistan    2020     1970.560169   2010            4.000000
...             ...     ...             ...    ...                 ...
108530     Zimbabwe    1990     2652.129209   1997            0.033080
108531     Zimbabwe    1990     2652.129209   1996            0.016790
108532     Zimbabwe    1990     2652.129209   1995            0.007684
108533     Zimbabwe    1990     2652.129209   1994            0.001739
108534     Zimbabwe    1990     2652.129209   1990            0.000000

[108535 rows x 5 columns]

Look at first and last 5 rows of dataframe:

In [58]:
gdp_and_internet_use.head()


Country  Year_x  GDP_Per_Capita Year_y  Internet_Users_Pct
0  Afghanistan    2020     1970.560169   2014            6.390000
1  Afghanistan    2020     1970.560169   2013            5.900000
2  Afghanistan    2020     1970.560169   2012            5.454545
3  Afghanistan    2020     1970.560169   2011            5.000000
4  Afghanistan    2020     1970.560169   2010            4.000000

In [54]:
gdp_and_internet_use.tail()

Country  Year_x  GDP_Per_Capita Year_y  Internet_Users_Pct
108530  Zimbabwe    1990     2652.129209   1997            0.033080
108531  Zimbabwe    1990     2652.129209   1996            0.016790
108532  Zimbabwe    1990     2652.129209   1995            0.007684
108533  Zimbabwe    1990     2652.129209   1994            0.001739
108534  Zimbabwe    1990     2652.129209   1990            0.000000

Subset the combined DataFrame to keep only the data for 2004, 2009, and 2014. Check that this happened correctly.

In [61]:
gdp_and_internet_use['Year_x'] == '2004'


0         False
1         False
2         False
3         False
4         False
          ...  
108530    False
108531    False
108532    False
108533    False
108534    False
Name: Year_x, Length: 108535, dtype: bool